tests/cupy_tests/math_tests/test_rational.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+
+import pytest
+
+import cupy
+from cupy import testing
+
+
+@testing.gpu
+class TestRational(unittest.TestCase):
+
+    @testing.for_int_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal()
+    def test_gcd_basic(self, xp, dtype):
+        a = testing.shaped_random((6, 6), xp, dtype, seed=0)
+        b = testing.shaped_random((6, 6), xp, dtype, seed=1)
+        return xp.gcd(a, b)
+
+    @testing.for_dtypes(['e', 'f', 'd', 'F', 'D'])
+    def test_gcd_dtype_check(self, dtype):
+        a = cupy.random.randint(-10, 10, size=(10, 10)).astype(dtype)
+        b = cupy.random.randint(-10, 10, size=(10, 10)).astype(dtype)
+        with pytest.raises(TypeError):
+            cupy.gcd(a, b)
+
+    @testing.for_int_dtypes(no_bool=True)
+    @testing.numpy_cupy_array_equal()
+    def test_lcm_basic(self, xp, dtype):
+        a = testing.shaped_random((6, 6), xp, dtype, seed=2)
+        b = testing.shaped_random((6, 6), xp, dtype, seed=3)
+        return xp.lcm(a, b)
```

Comment:
`lcm(0, 0)` is not tested in this seed. Why not avoid using `shaped_random` and write a test case explicitly?.

Reply:
I have added additional test cases that cover boundary cases -> (0,0), negative and positive, and negative and negative

Reply:
What is the role of `test_lcm_basic`? I think it is better to write all the test cases explicitly.

Reply:
I have added the test cases with all values explicitly written and have tried to choose values that represent all the boundary cases.